### ISSUE TITLE
[BUG] Deleting duplicate pictures lead to negative remainder counter

### ISSUE THREAD

**Issue by ytzelf** [immich-app/immich#968]
**Describe the bug**
After adding the Camera album on my Android phone to Immich I deleted numerous pictures from another app (duplicate finder). The assets remained in the cloud and the dashboard now show a negative Remainder.

**To Reproduce**
1. Add an album to Backup Albums on the mobile app with numerous duplicate pictures
2. Wait for backup to be finished
3. Batch delete duplicate pictures in that album from a 3rd party app
4. Pictures are not on the phone anymore but still backed up leading to a negative Remainder counter

**Expected behavior**
Some kind of purging mechanism where pictures that don't live on the phone are deleted from the server.

**Screenshots**

**System**
- Phone OS Android: `12`
- Server Version: `1.35`
- Mobile App Version: `1.35`

**Additional context**

**Comment by alextran1502** [comment]
confirmed

**Comment by alextran1502** [comment]
@ytzelf I don't think you want that because then if you remove photos or videos on the phone to have more space to take more photos they will be also deleted on the server. Which is not what backing up means.

**Comment by ytzelf** [comment]
OK then i'm not sure how to work around that issue.

What I could see working is 
1/ Select all items in backups and not on the phone (separate issue - easily selecting all items, items for a specific date range or items based on their storage location is needed)
2/ Delete these items on the backup location from the mobile app (maybe a delete subdialog on pictures - "delete from phone" and "delete from backup")

If you're fine having pictures on backup that were explicitely deleted from the phone then I'm not sure this is even an issue at all tbh (and the remainder counter can just be max(0, total-uploaded)). Probably more a matter of being able to view, quickly select, and delete quickly these pictures.

**Comment by gh3597** [comment]
May be related to this issue but I got a negative number after closing and re-opening the app, no deleting needed.

**Comment by spupuz** [comment]
Same proplem for me.

**Comment by SpecialAro** [comment]
> May be related to this issue but I got a negative number after closing and re-opening the app, no deleting needed.

I just left my phone to do it's first backup on a freshly new immich server during the night and I got this.

![Screenshot_20221221-074424](https://user-images.githubusercontent.com/37463445/208849282-d7ca8aab-4b1d-44c1-a292-170f7a8f5c1f.jpg)

**Comment by alextran1502** [comment]
> > May be related to this issue but I got a negative number after closing and re-opening the app, no deleting needed.
> 
> 
> 
> I just left my phone to do it's first backup on a freshly new immich server during the night and I got this.
> 
> 
> 
> ![Screenshot_20221221-074424](https://user-images.githubusercontent.com/37463445/208849282-d7ca8aab-4b1d-44c1-a292-170f7a8f5c1f.jpg)
> 
> 

Beside backup up from the phone, did by chance upload anything else from the web or from the CLI? If you go on the web and hover on the question mark next to the Photos button, what is the sum of the photos and videos count?

**Comment by SpecialAro** [comment]
> > > May be related to this issue but I got a negative number after closing and re-opening the app, no deleting needed.
> >
> 
> Beside backup up from the phone, did by chance upload anything else from the web or from the CLI?

Nop, just through this phone.

> If you go on the web and hover on the question mark next to the Photos button, what is the sum of the photos and videos count?

[image](https://user-images.githubusercontent.com/37463445/208922050-9ea8b069-b1a7-49ed-86f2-eea7036350eb.png)
This?

**Comment by jrasm91** [comment]
Is this related to Live Photos being stored as two parts (and showing up in the count as two separate assets)? Is there a way for you to easily tell how many live photos you have?

Ideally the backup count should be how many assets are left to backup, which is unrelated to how many are on the server or have previously been backed-up to the server. It should be something like: `count(assets from selected albums on device) - count(where on device asset is in list of assets on the server)`.

**Comment by gh3597** [comment]
> Is this related to Live Photos being stored as two parts 

Wouldn't have thought so as I have zero live photos.

All I did when I got this initially was to install the app and let it sync. No other upload methods used at that point in time.

**Comment by alextran1502** [comment]
> > Is this related to Live Photos being stored as two parts
> 
> Wouldn't have thought so as I have zero live photos.
> 
> All I did when I got this initially was to install the app and let it sync. No other upload methods used at that point in time.

I think you have some duplicate photos or videos and the app doesn't handle the duplication count correctly.

**Comment by SpecialAro** [comment]
> > > Is this related to Live Photos being stored as two parts
> > 
> > 
> > Wouldn't have thought so as I have zero live photos.
> > All I did when I got this initially was to install the app and let it sync. No other upload methods used at that point in time.
> 
> I think you have some duplicate photos or videos and the app doesn't handle the duplication count correctly. Can you help me confirm if you have some duplicate photos or videos in your library?

Just quickly looked through all my photos and I can't find any duplicates (at least the file names are always different).

**Comment by gh3597** [comment]
> I think you have some duplicate photos or videos and the app doesn't handle the duplication count correctly. Can you help me confirm if you have some duplicate photos or videos in your library?

Not as far as I'm aware. Depends on what the criteria is? File names will all be unique, anything "duplicate" will be things like downloaded items that have no or minimal metadata?

**Comment by samip5** [comment]
I have also noticed this issue on Android.

![Screenshot_20230128-201856.png](https://user-images.githubusercontent.com/1703002/215284390-f6580b2e-1fdd-4f74-81f9-59a50415f435.png)

**Comment by ikaruswill** [comment]
Dropping by to share a finding. 

- I noticed that on my partner's device, the negative count `-1` coincides with the exact number of files stuck in the `upload` subfolder (which was corrupted due to incomplete/failed upload). 
- However that number for my device `-30` is still much larger than the number of stuck files `5`.

My hypothesis is that there is a possibility that files stuck in the `upload` folder may count towards library size and therefore contribute towards this negative photo count issue.

**Comment by bigbeka** [comment]
Using iOS client:

I understand that the counter just subtracts 'Backup' number from 'Total' that's on device.

My suggestion would be to compare the last backed up photo/video ID with the one on device and then count from there how many more photos/videos are to be backed up.

**Comment by nocomputeruser** [comment]
Wanted to leave a note that I just experienced this as well on iOS. Not sure if you need me to help gather information or not for you.

**Comment by mtettke123** [comment]
Seem to have the same problem:

![Screenshot_2023-06-03-16-49-36-682_app alextran immich](https://github.com/immich-app/immich/assets/82048170/465c77a4-00f4-4602-a9cb-4978f2519b3b)

Workflow: I started by uploading images from my new phone using immich - after deciding it's a great tool I installed immich on my old phone to upload these pictures, too. Maybe that's the problem?! ...

**Comment by mmorys** [comment]
Same issue. Fresh setup and was backing up ~8000 photos/videos on my Android phone. I had made changes during the process, adding 2 more albums, and restarting my phone. At one point the Total started showing ~5000 even though camera still had 8000 assets.

I was able to resolve the issue by deleting the app's cache and data. Upon login to a fresh app, I reselected the backup albums and the Total is now correct, matching Backup.

**Comment by d3473r** [comment]
> Same issue. Fresh setup and was backing up ~8000 photos/videos on my Android phone. I had made changes during the process, adding 2 more albums, and restarting my phone. At one point the Total started showing ~5000 even though camera still had 8000 assets.
> 
> I was able to resolve the issue by deleting the app's cache and data. Upon login to a fresh app, I reselected the backup albums and the Total is now correct, matching Backup. 

Thank you, this workaround fixes the issue for me also :)

**Comment by ytzelf** [comment]
> Same issue. Fresh setup and was backing up ~8000 photos/videos on my Android phone. I had made changes during the process, adding 2 more albums, and restarting my phone. At one point the Total started showing ~5000 even though camera still had 8000 assets.
> 
> I was able to resolve the issue by deleting the app's cache and data. Upon login to a fresh app, I reselected the backup albums and the Total is now correct, matching Backup.

That ... did it for me as well. Thanks a lot it was somewhat irking ^^

**Comment by NextBlaubeere** [comment]
I freshly installed immich two times, to see if the error was on my side, but i can confirm this bug:

1) Install Immich server
2) Install Android app
3) Select "Camera" Folder to upload (not much, ~400 Photos)
4) Let the phone sit for a while to finish
5) Encounter this bug (numbers seem to differ each installation)

![Screenshot_20230831-174226_Immich(1)](https://github.com/immich-app/immich/assets/10467690/410c0ac2-aef4-451a-a82c-af6fbd9a2a19)

**Comment by lead0r** [comment]
I can confirm that this does happen with the Android app with a freshly installed server and that deleting app cache and data doesn't really solve the issue

**Comment by jrnewell** [comment]
I was able to reproduce this following these steps (I am on ios).

1. Have auto-upload in foreground on
2. Start app and let it upload new pictures
3. Don't migrate away from Photos tab, go to one of the new pictures that was uploaded.  You can still click the "upload asset" button.  Click it and confirm upload
4. The asset will be uploaded again and now the remaining will be "negative" indefinitely.

**Comment by vitor-alves** [comment]
Deleting app data and cached on Android worked for me

**Comment by Tyree** [comment]
I have the same problem. Mine is -1677.
I have theorized that what caused it was initially having the "Recent" category selected when I enabled auto-backup. Items in that category will overlap items in other selected categories/locations. So once I realized and deselected it, there were already "extra" items uploaded that the app now sees as a negative count. More items auto-uploaded than the app is set for.
